src/goose.rs

```diff
@@ -1128,7 +1131,7 @@ impl GooseUser {
         }
 
         // Otherwise use the `base_url`.
-        Ok(self.base_url.join(path)?.to_string())
+        Ok(self.base_url.as_ref().unwrap().join(path)?.to_string())
```

Comment:
Could we check for the `None` case here and explicitly return an error instead of panicking?

Maybe use https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or and then `?` to return the error? This way we do not introduce a new means of panicking the processes.

Reply:
Let me know what you think of the update.